### PR TITLE
Make analysis success not only dependent on flag but also on error list.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/6019

### CODE REVIEW

File: libyul/AsmAnalysis.cpp [hunk 1]
```diff
@@ -52,7 +52,8 @@ bool AsmAnalyzer::analyze(Block const& _block)
 	if (!(ScopeFiller(m_info, m_errorReporter))(_block))
 		return false;
 
-	return (*this)(_block);
+	bool success = (*this)(_block);
+	return success && !m_errorReporter.hasErrors();
```
Comment (leonardoalt) [inline libyul/AsmAnalysis.cpp:58]: Is it guaranteed that `exception -> m_errorReporter.hasErrors()`?

Comment (chriseth) [reply libyul/AsmAnalysis.cpp:58]: What do you mean by `execption`?

The problem was that this component found an error but due to the internal logic, it was difficult to return `false` at that point. I changed it to just also include into the condition any errors found but not explicitly flagged by return.

Comment (axic) [reply libyul/AsmAnalysis.cpp:58]: Separately, do we need an assertion for `!success && hasErrors()` ?

Comment (chriseth) [reply libyul/AsmAnalysis.cpp:58]: no, this fixes a bug where we returned success but there were errors, so the assertion will trigger.

**Review by leonardoalt — DISMISSED** [review]

**Review by axic — APPROVED** [review]

**Pull request merged**